src/algebra/order/to_interval_mod.lean

```diff
@@ -181,69 +181,111 @@ by { rw [to_Ioc_mod_eq_iff hp, set.right_mem_Ioc], exact ⟨lt_add_of_pos_right
 to_Ico_div_eq_of_sub_zsmul_mem_Ico hp $
   by simpa only [add_smul, add_sub_add_right_eq_sub] using sub_to_Ico_div_zsmul_mem_Ico hp a b
 
+@[simp] lemma to_Ico_div_add_zsmul' (a b: α) (m : ℤ) :
+  to_Ico_div hp (a + m • p) b = to_Ico_div hp a b - m :=
+begin
+  refine to_Ico_div_eq_of_sub_zsmul_mem_Ico _ _,
+  rw [sub_smul, ←sub_add, add_right_comm],
+  simpa using sub_to_Ico_div_zsmul_mem_Ico hp a b,
+end
+
 @[simp] lemma to_Ioc_div_add_zsmul (a b : α) (m : ℤ) :
   to_Ioc_div hp a (b + m • p) = to_Ioc_div hp a b + m :=
 to_Ioc_div_eq_of_sub_zsmul_mem_Ioc hp $
   by simpa only [add_smul, add_sub_add_right_eq_sub] using sub_to_Ioc_div_zsmul_mem_Ioc hp a b
 
+@[simp] lemma to_Ioc_div_add_zsmul' (a b : α) (m : ℤ) :
+  to_Ioc_div hp (a + m • p) b = to_Ioc_div hp a b - m :=
+begin
+  refine to_Ioc_div_eq_of_sub_zsmul_mem_Ioc _ _,
+  rw [sub_smul, ←sub_add, add_right_comm],
+  simpa using sub_to_Ioc_div_zsmul_mem_Ioc hp a b,
+end
+
 @[simp] lemma to_Ico_div_zsmul_add (a b : α) (m : ℤ) :
   to_Ico_div hp a (m • p + b) = m + to_Ico_div hp a b :=
 by rw [add_comm, to_Ico_div_add_zsmul, add_comm]
 
 @[simp] lemma to_Ioc_div_zsmul_add (a b : α) (m : ℤ) :
-  to_Ioc_div hp a (m • p + b) = to_Ioc_div hp a b + m :=
+  to_Ioc_div hp a (m • p + b) = m + to_Ioc_div hp a b :=
 by rw [add_comm, to_Ioc_div_add_zsmul, add_comm]
```

Comment:
You're missing the new variants of those two

Reply:
What statement are you looking for? `to_Ioc_div hp a (m • p + b)  = -m + to_Ioc_div hp a b`? Negation does't seem too pleasant there.

Reply:
I'm thinking of `to_Ioc_div hp (m • p + a) b = to_Ioc_div hp a b - m`

Reply:
I don't like that because it commutes the side that `m` operates on the result. If you're going to swap the order, you may as well just `rw add_comm` yourself as the caller.

Reply:
Then delete `to_Ico_div_zsmul_add`/`to_Ioc_div_zsmul_add`.

Reply:
I don't follow; `to_Ico_div_zsmul_add` and `to_Ioc_div_zsmul_add` start and end with `m` on the left of the binary operator.
`to_Ico_div_add_zsmul` and `to_Ioc_div_add_zsmul` start and end with `m` on the right of the binary operator.

The lemma you're proposing starts wiith `m` on the left but ends up with it on the right, which is what I object to.

Reply:
Still, people could think it's missing, hence why I'm suggesting adding a note.

Reply:
Hey, you cheated and edited your comment!

Reply:
Less than a minute having posted it 😆

Reply:
Added a comment.